src/dsp_tools/commands/xml_validate/sparql/value_shacl.py

```diff
@@ -0,0 +1,135 @@
+from rdflib import Graph
+
+
+def construct_property_shapes(onto: Graph) -> Graph:
+    """
+    Returns the sh:PropertyShape for the properties in the ontology.
+
+    Args:
+        onto: ontology graph
+
+    Returns:
+        Graph with the property shapes
+    """
+    g = Graph()
+    g += _construct_property_type_shape_based_on_object_type(onto)
+    g += _construct_property_type_shape_based_on_gui_element(onto)
+    return g + _add_property_shapes_to_class_shapes(onto)
+
+
+def _add_property_shapes_to_class_shapes(onto: Graph) -> Graph:
+    query_s = """
+    PREFIX owl: <http://www.w3.org/2002/07/owl#> 
+    PREFIX sh: <http://www.w3.org/ns/shacl#>
+    PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+    PREFIX salsah-gui: <http://api.knora.org/ontology/salsah-gui/v2#> 
+    PREFIX api-shapes: <http://api.knora.org/ontology/knora-api/shapes/v2#>
+    PREFIX knora-api:  <http://api.knora.org/ontology/knora-api/v2#>
+
+    CONSTRUCT {
+
+      ?classShapeIRI sh:property ?propShapesIRI .
+
+    } WHERE {
+
+      ?class a owl:Class ;
+          knora-api:isResourceClass true ;
+          knora-api:canBeInstantiated true ;
+          rdfs:subClassOf ?restriction .
+      ?restriction a owl:Restriction ;
+          owl:onProperty ?propRestriction ;
+          salsah-gui:guiOrder ?order .
+      FILTER NOT EXISTS { ?propRestriction knora-api:isLinkValueProperty true }
+
+      BIND(IRI(CONCAT(str(?class), "_Shape")) AS ?classShapeIRI)
+      BIND(IRI(CONCAT(str(?propRestriction), "_PropShape")) AS ?propShapesIRI)
+    }
+    """
+    if results_graph := onto.query(query_s).graph:
+        return results_graph
+    return Graph()
+
+
+def _construct_property_type_shape_based_on_object_type(onto: Graph) -> Graph:
+    property_type_mapper = {
+        "knora-api:BooleanValue": "api-shapes:BooleanValue_ClassShape",
+        "knora-api:ColorValue": "api-shapes:ColorValue_ClassShape",
+        "knora-api:DateValue": "api-shapes:DateValue_ClassShape",
+        "knora-api:DecimalValue": "api-shapes:DecimalValue_ClassShape",
+        "knora-api:GeonameValue": "api-shapes:GeonameValue_ClassShape",
+        "knora-api:IntValue": "api-shapes:IntValue_ClassShape",
+        "knora-api:ListValue": "api-shapes:ListValue_ClassShape",
+        "knora-api:TimeValue": "api-shapes:TimeValue_ClassShape",
+        "knora-api:UriValue": "api-shapes:UriValue_ClassShape",
+    }
+    g = Graph()
+    for object_type, shacl_shape in property_type_mapper.items():
+        g += _construct_one_property_type_shape_based_on_object_type(onto, object_type, shacl_shape)
+    return g
```

Comment:
*thought*: There seems to be a naming convention hidden in the `property_type_mapper`

IMHO it is worth expressing this naming convention explicitly.

For example by adding an inner function:
```python
def _construct_property_type_shape_based_on_object_type(onto: Graph) -> Graph:
    def as_object_type_and_shacl_shape(property_type: str) -> tuple[str, str]:
        return "knora-api:" + property_type + "Value", "api-shapes:" + property_type + "Value_ClassShape"

    property_types = {"Boolean", "Color", "Date", "Decimal", "Geoname", "Int", "List", "Time", "Uri"}
    g = Graph()
    for t in property_types:
        object_type, shacl_shape = as_object_type_and_shacl_shape(t)
        g += _construct_one_property_type_shape_based_on_object_type(onto, object_type, shacl_shape)
    return g
```